go-controller/pkg/util/pod_annotation.go

```diff
@@ -87,6 +87,10 @@ type podRoute struct {
 // MarshalPodAnnotation returns a JSON-formatted annotation describing the pod's
 // network details
 func MarshalPodAnnotation(podInfo *PodAnnotation) (map[string]string, error) {
+	err := verifyPodAnnotation(podInfo)
```

Comment:
So what happens in the case where we pass in `[]net.IP{nil}` for `Gateways`? It seems to me that we'd end up marshalling it to `"gateway_ips": ["<nil>"]` in the pod annotation, but then it would fail to parse on the other side in `UnmarshalPodAnnotation` and so we'd fail there, right? I think that's a suitable way to fail in this case. This isn't something an end-user should ever see.

If we *didn't* get an error on the unmarshalling side, or if that error didn't cause any obvious failure, then I think that's what we need to fix.

Reply:
you are right about the error on the unmarshalling side. Someone didn't check the return from net.ParseIP() correctly. I changed it and updated the test case and removed the verify function